src/Feersum.CompilerServices/Diagnostics.fs

```diff
@@ -91,18 +94,30 @@ type Diagnostic =
         | Point p ->
             sprintf "%s(%d,%d): %s: %s" (p.Source |> normaliseName) p.Line p.Col d.MessagePrefix d.FormattedMessage
         | Span (s, e) ->
-            sprintf
-                "%s(%d,%d,%d,%d): %s: %s"
-                (s.Source |> normaliseName)
-                s.Line
-                s.Col
-                e.Line
-                e.Col
-                d.MessagePrefix
-                d.FormattedMessage
+            // If both points are on the same line then we can use the a more
+            // compact format.
+            if s.Line = e.Line then
+                sprintf
+                    "%s(%d,%d-%d): %s: %s"
+                    (s.Source |> normaliseName)
```

Comment:
This breaks click through and problem matching in VS Code. Maybe Code needs a PR?

Reply:
https://github.com/microsoft/vscode/pull/154054